### PR TITLE
Allow for GUIs to be run at 60 FPS

### PR DESCRIPTION
**Pull request by bruberu**
There have been complaints about some GUIs, mainly in the terminal being too laggy, and I traced this back to the fact that currently, all GUIs update at one tick per second, or 20 FPS, which is noticeably choppy to most humans. This fixes that issue by allowing widgets to access an updater at 60 FPS through updateScreenOnFrame. This is applied to AnimaWidgetGroup and PongApp so far, although others can be moved over at request.

### CODE REVIEW

File: src/main/java/gregtech/api/gui/impl/ModularUIGui.java [hunk 1]
```diff
@@ -72,6 +72,11 @@ public void handleWidgetUpdate(PacketUIWidgetUpdate packet) {
 
     @Override
     public void drawScreen(int mouseX, int mouseY, float partialTicks) {
+        timeSinceUpdate += partialTicks;
+        if(timeSinceUpdate > 0.3) {
+            this.timeSinceUpdate -= 0.3;
+            modularUI.guiWidgets.values().forEach(Widget::updateScreenOnFrame);
+        }
```
Comment (Yefancy) [inline src/main/java/gregtech/api/gui/impl/ModularUIGui.java:79]: why it is 60fps.

```
long now = System.currentTimeMillis();
if (now - last >= 17) {
    last = now;
    modularUI.guiWidgets.values().forEach(Widget::updateScreenOnFrame);
}
```

Comment (bruberu) [reply src/main/java/gregtech/api/gui/impl/ModularUIGui.java:79]: It approximates a third of a tick, making it 60 FPS. Also, didn't you want this to use partialTicks?

Comment (Yefancy) [reply src/main/java/gregtech/api/gui/impl/ModularUIGui.java:79]: but idk whats your logic (`timeSinceUpdate += partialTicks;`)
Can you explain it to me? Let's say the current FPS is 120 (video settings). 
Ideally the period for partialsTicks in some case would be like this:
 (1/6=> 2/6=>3/6=>4/6=>5/6=>6/6) hit in one tick, but your logic here will be hit 4 times (3,4,5,6) (80fps). 
another example, also Ideally the period for partialsTicks but different initial case:
 (0/6=>1/6=> 2/6=>3/6=>4/6=>5/6), your logic will be hit 3 times (3,4,5) (60fps).
In fact, partialsTicks are not so ideal, the FPS you simulate can be higher or smaller depending on the FPS of the game, not 60.

imo, using partialsTicks should be:
```
float now = getModularUI().entityPlayer.ticksExisted + partialTicks;
if (now - last >= 0.333) {
    last = now;
    modularUI.guiWidgets.values().forEach(Widget::updateScreenOnFrame);
}
```
If this is my math problem, please ignore this comment. I just scribbled the math in my head, not in detail to calculate the expectation or your logic.

**Comment by Yefancy** [comment]
I made a minor change. There was a small issue previous, when the FPS was low (settings or rendering performance) it would run missing some frame loops. Missing frame loops are now executed even when the FPS is low, rather than skipping them.

**Review by Yefancy — APPROVED** [review]

**Pull request merged**